src/main/java/no/unit/nva/doi/transformer/DataciteResponseConverter.java

```diff
@@ -93,10 +120,117 @@ private String createDescription() {
+                .withEnd(container.getLastPage())
+                .build();
+    }
+
+    private PublicationContext extractPublicationContext(DataciteResponse dataciteResponse) throws
+            InvalidIssnException {
+        PublicationType type = getPublicationType(dataciteResponse);
+        if (nonNull(type) && type.equals(JOURNAL_CONTENT)) {
+
+            return new Journal.Builder()
+                    .withPrintIssn(extractPrintIssn(dataciteResponse))
+                    .withTitle(dataciteResponse.getContainer().getTitle())
+                    .withOnlineIssn(extractOnlineIssn(dataciteResponse))
+                    .withPeerReviewed(true)
+                    .withOpenAccess(extractOpenAccess(dataciteResponse))
+                    .withLevel(Level.NO_LEVEL)
+                    .build();
+        }
         return null;
     }
 
+    private String extractOnlineIssn(DataciteResponse dataciteResponse) {
+        DataciteRelatedIdentifier result = extractPartOfRelations(dataciteResponse)
+                .stream()
+                .filter(this::isOnlineIssn)
+                .collect(SingletonCollector.collectOrElse(null));
+        return nonNull(result) ? IssnCleaner.clean(result.getRelatedIdentifier()) : null;
+
+    }
+
+    private boolean isOnlineIssn(DataciteRelatedIdentifier identifier) {
+        return DataciteRelatedIdentifierType.getByCode(identifier.getRelatedIdentifierType())
+                .equals(DataciteRelatedIdentifierType.EISSN);
+    }
+
+    private String extractPrintIssn(DataciteResponse dataciteResponse) {
+        DataciteRelatedIdentifier result = extractPartOfRelations(dataciteResponse)
+                .stream()
+                .filter(this::isPrintIssn)
+                .collect(SingletonCollector.collectOrElse(null));
+        return nonNull(result) ? IssnCleaner.clean(result.getRelatedIdentifier()) : null;
+    }
+
+    private List<DataciteRelatedIdentifier> extractPartOfRelations(DataciteResponse dataciteResponse) {
+        return dataciteResponse.getRelatedIdentifiers()
+                .stream()
+                .filter(this::isPartOf)
+                .collect(Collectors.toList());
+    }
+
+    private boolean isPrintIssn(DataciteRelatedIdentifier identifier) {
+        return DataciteRelatedIdentifierType.getByCode(identifier.getRelatedIdentifierType())
+                .equals(DataciteRelatedIdentifierType.ISSN);
+    }
+
+    private boolean isPartOf(DataciteRelatedIdentifier identifer) {
+        return DataciteRelationType.getByRelation(identifer.getRelationType())
+                .equals(DataciteRelationType.IS_PART_OF);
+    }
+
+    private boolean extractOpenAccess(DataciteResponse dataciteResponse) {
+        List<DataciteRights> rights = Optional.ofNullable(dataciteResponse.getRightsList()).orElse(null);
+        if (nonNull(rights)) {
+            return rights.stream().anyMatch(this::hasOpenAccessRights);
+        }
+        return false;
+    }
```

Comment:
Line 218 is practically dataciteResponse.getRIghtsList(); Probably what you want is
`Optional.ofNullable(dataciteResponse).map(d->d.getRightsList).orElse(null);`

A more compact but less readable in my opinion version of this function is the following: 

```suggestion
      return Optional.ofNullable(dataciteResponse)
            .map(DataciteResponse::getRightsList)
            .stream()
            .flatMap(d -> d.stream())
            .anyMatch(this::hasOpenAccessRights);
```